### ISSUE TITLE
Sentence zero contains 21,723 different sentences

### ISSUE THREAD

**Issue by mathematiguy** [TeHikuMedia/corpora#202]
I've been looking through different kinds of duplicates in the json data, and one thing I noticed was that records with `sentence_id==0` seem to contain arbitrary sentences. At the moment, it seems there are over 20,000 distinct sentences with sentence id 0.

Ideally these sentence ids would be recalculated somehow - it seems corpora is the correct place to do this rather than in waha-tuhi.

**Comment by edwardabraham** [comment]
Eek! Good sleuthing, Caleb.

**Comment by mathematiguy** [comment]
This could be a related problem or a different one, but I've got around 42,000 records in the corpora json with an empty `sentence` field (which we call usually `sentence_id`), but which do have a `sentence_text` field.

Here are some examples:
```
{
  "person": 710708,
  "sentence": null,
  "audio_file_url": "https://koreromaori.com/recording-file/224075/",
  "id": 224075,
  "sentence_text": "Nā <c.kirehu>koia tēnā</c>, ā, ka waihotia ki reira.",
  "updated": "2020-05-30T01:01:03.961470+12:00",
  "audio_file_md5": "27caa7b0fadc7888f7997c9fa9468bea",
  "audio_file_wav_md5": "33f3c4c11def25cd137d131bf4ae6045",
  "private": true,
  "transcription": {
    "text": "tēnā koe tēnā ā ko ia te kite",
    "wer": 0.558139534883721
  },
  "quality_control_aggregate": null
}
{
  "person": 710638,
  "sentence": null,
  "audio_file_url": "https://koreromaori.com/recording-file/219211/",
  "id": 219211,
  "sentence_text": "tēnā anō koutou katoā e whakarongo mai nei",
  "updated": "2020-05-27T19:08:49.594677+12:00",
  "audio_file_md5": "c0b3dae5014af05f54ddbb797acecabe",
  "audio_file_wav_md5": "75b4f743876fdc7b1e3521eac0c44235",
  "private": true,
  "transcription": {
    "text": "Could not get a transcription. ReadTimeout",
    "wer": 0.880952380952381
  },
  "quality_control_aggregate": null
}
{
  "person": 493222,
  "sentence": null,
  "audio_file_url": "https://koreromaori.com/recording-file/200339/",
  "id": 200339,
  "sentence_text": "Engari rā, ko ētahi e kī ana, kāore e tika kia kaha te kori o te hope. Āe rā hoki.",
  "updated": "2019-08-15T21:10:11.845137+12:00",
  "audio_file_md5": "26b3b13e4b33b9598aff4e6f097dd9ac",
  "audio_file_wav_md5": "508a87b0f7a7d5d027a7e171829914c6",
  "private": true,
  "transcription": {
    "text": "engari ko ētahi ki ana kaitekateka te kore o te rā",
    "wer": 0.448717948717949
  },
  "quality_control_aggregate": null
}
```

**Comment by edwardabraham** [comment]
Yep, I can confirm (I am playing with the Django Queryset API from a python notebook running on the staging server):
```
> Recording.objects.filter(sentence=None).count()
41106
```

**Comment by mathematiguy** [comment]
Just checking the IDs more closely, on my version of the json data, in the `recording_fetch` table we've got:
- 126 records with missing speaker ids (`speaker`)
- 42241 records with missing sentence ids (`sentence`)
- 104 sentences with missing text (`sentence_text`)

And no missing recording IDs (`id`), which is good because this would be especially weird/bad

**Comment by edwardabraham** [comment]
I think that the records with missing sentences are the ones that have been imported from Kaituhi, so that they have a sentence_text, but because they are not read from within Kōrero Maori, they don't have a sentence id.

**Comment by edwardabraham** [comment]
A null sentence would is probably the correct way to treat this? Another way would be to add the sentences to the sentence data.  This is a bigger question on ow we integrate data from Kaituhi (and ultimately from other sources such as the Whare Kōrero, into the Kōrero Māori project).

**Comment by edwardabraham** [comment]
I am interested in the difference between the number from staging (41106) and your number (42241) @mathematiguy ?

**Comment by mathematiguy** [comment]
So in that case, I should probably keep the existing behaviour and map those missing sentence IDs to speaker zero and keep going?

> I am interested in the difference between the number from staging (41106) and your number (42241) @mathematiguy ?

I'm using the data we grabbed from gorbachev in this job on Feb 1st: https://gorbachev.koreromaori.io/#/job/waha-tuhi-fetch/1894 

I don't have a hot connection to the corpora database, so I just use what I have at the time. I am a little surprised that my number is larger than yours, but maybe there was some kind of update between now and then.

**Comment by mathematiguy** [comment]
As other datasets get added to corpora, if they all get null sentence IDs we won't be able to reverse engineer what came from where will we? Maybe we don't need to mind, but it's something to consider I guess.

**Comment by edwardabraham** [comment]
I think that we need to record the source of the data. Important that we are able to re-upload  without getting more duplicates. I note that  the person association with the last one of your example recordings is `kaituhi-transcript-100-Cissy Midtgard`

**Comment by edwardabraham** [comment]
The "kaituhi-transcript-100" shows that the current schema is a bit lacking!
